feature/google/src/test/java/com/revenuecat/purchases/google/BillingWrapperTest.kt

```diff
@@ -315,6 +317,94 @@ class BillingWrapperTest {
         )
     }
 
+    @Test
+    fun `obfuscatedAccountId is set for non-transfer purchases`() {
+        setup()
+        val appUserID = "jerry"
+
+        val skuDetails = stubSkuDetails(productId = "product_a")
+        val activity: Activity = mockk()
+
+        mockkStatic(BillingFlowParams::class)
+        val mockBuilder = mockk<BillingFlowParams.Builder>(relaxed = true)
+        every {
+            BillingFlowParams.newBuilder()
+        } returns mockBuilder
+
+        every {
+            mockBuilder.setSkuDetails(any())
+        } returns mockBuilder
+
+        val params = mockk<BillingFlowParams>(relaxed = true)
+        every {
+            mockBuilder.build()
+        } returns params
+
+        every {
+            mockClient.launchBillingFlow(any(), params)
+        } returns BillingClient.BillingResponseCode.OK.buildResult()
+
+        billingClientStateListener!!.onBillingSetupFinished(BillingClient.BillingResponseCode.OK.buildResult())
+
+        wrapper.makePurchaseAsync(
+            activity,
+            appUserID,
+            skuDetails.toProductDetails(),
+            null,
+            null
+        )
+
+        verify {
+            mockBuilder.setObfuscatedAccountId(appUserID.sha256())
+        }
+
+        clearStaticMockk(BillingFlowParams::class)
```

Comment:
this is definitely the lazy way out... 

looks like mocking the BillingFlowParams.Builder with relaxed = true was causing `properlySetsBillingFlowParams` to fail -- the params were always all null. mocking a builder seemed tricky, but that could be due to my lack of mockk experience... could be useful in the future, but for these tests didn't seem necessary since we really only verify that a method was called.

Reply:
maybe this should be a part of tearDown?

Reply:
I don't inherently see a problem with just mocking the whole thing for a test and clearing the mock at the end, especially if it makes the test code simpler and more focused than mocking everything in it. 

The only advantage I'd see is that with explicit mocks per-method we'd make sure that we're only calling what we need, but it isn't the focus for this particular test anyway.

Reply:
@aboedo junit only provides @After (for after every test, which might slow things?) or @AfterClass for after all tests.. the other option would be to extract these two tests into their own class, but that seems like overkill. let me know what you think

Reply:
not sure what's best practice here tbh @vegaro thoughts?

Reply:
Since we are not calling `mockkStatic` in `setup` (@Before) I don't think we should call `clearStaticMockk` in `teardDown` (@After). I think having it in each test makes sense. Maybe since we have a `setUpForObfuscatedAccountIDTests` we could do a `tearDownForObfuscatedAccountIDTests` that does the clear. 

With JUnit5 (we use JUnit4 still), we could have inner classes that would solve this issue. `purchases-hybrid-common` already uses JUnit5

Reply:
@vegaro want me to make a clubhouse ticket for upgrading to junit5? didn't see one with a quick search